dart/common/StlHelpers.h

```diff
@@ -70,6 +71,21 @@ static T getVectorObjectIfAvailable(size_t index, const std::vector<T>& vec)
   return nullptr;
 }
 
+//==============================================================================
+/// static_if_else allows the compiler to choose between two different possible
+/// types at runtime based on a runtime boolean.
+template <bool B, typename T_if, typename T_else>
+struct static_if_else
```

Comment:
I think `static_if_else` is identical to [`std::conditional`](http://en.cppreference.com/w/cpp/types/conditional).